cmake/modules/FindPyArrow.cmake

```diff
@@ -24,37 +24,25 @@ execute_process(
           "from __future__ import print_function\ntry: import pyarrow; print(' '.join(pyarrow.get_libraries()), end='')\nexcept:pass"
   OUTPUT_VARIABLE __pyarrow_libraries)
 
-# And the version
-execute_process(
-  COMMAND "${Python_EXECUTABLE}" -c
-          "from __future__ import print_function\ntry: import pyarrow; print(pyarrow.__version__, end='')\nexcept:pass"
-  OUTPUT_VARIABLE __pyarrow_version)
-
   find_path(PYTHON_PYARROW_INCLUDE_DIR arrow/python/api.h
   HINTS "${__pyarrow_path}" "${PYTHON_INCLUDE_PATH}" NO_DEFAULT_PATH)
 
 set(PYTHON_PYARROW_LIBRARY_DIR ${__pyarrow_library_dirs})
 
-# Figure out the major version for the .so/.dylibs
-string(REPLACE "." ";" PYARROW_VERSION_LIST ${__pyarrow_version})
-list(GET PYARROW_VERSION_LIST 0 PYARROW_VERSION_MAJOR)
-list(GET PYARROW_VERSION_LIST 1 PYARROW_VERSION_MINOR)
-list(GET PYARROW_VERSION_LIST 2 PYARROW_VERSION_PATCH)
-
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
   # windows its just "arrow.dll"
   set(PYTHON_PYARROW_PYTHON_SHARED_LIBRARY "arrow_python")
   set(PYTHON_PYARROW_ARROW_SHARED_LIBRARY "arrow")
   set(PYTHON_PYARROW_LIBRARIES ${PYTHON_PYARROW_PYTHON_SHARED_LIBRARY} ${PYTHON_PYARROW_ARROW_SHARED_LIBRARY})
 elseif (CMAKE_SYSTEM_NAME MATCHES "Darwin")
   # Link against pre-built libarrow on MacOS
-  set(PYTHON_PYARROW_PYTHON_SHARED_LIBRARY ${PYTHON_PYARROW_LIBRARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX}arrow_python.${PYARROW_VERSION_MINOR}.dylib)
-  set(PYTHON_PYARROW_ARROW_SHARED_LIBRARY ${PYTHON_PYARROW_LIBRARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX}arrow.${PYARROW_VERSION_MINOR}.dylib)
+  set(PYTHON_PYARROW_PYTHON_SHARED_LIBRARY ${PYTHON_PYARROW_LIBRARY_DIR}/${CMAKE_SHARED_LIBRARY_PREFIX}arrow_python.100.dylib)
```

Comment:
We are going to have to cope with the conditional naming for these libraries, as they are not disted in this version under this name (and don't seem to exhibit any consistent version convention at all).

Reply:
Let's link this into an apache arrow issue, for pyarrow < 1 they had symlinks and we'll need those for some platforms

Reply:
https://github.com/apache/arrow/pull/7334

Reply:
@wesm this makes linking against these libraries super ugly. Since the version is managed by python anyway (e.g. in the version of the python package), wouldn't it be better to ship just the unversioned binary (e.g. libarrow.dylib)?

Reply:
See http://arrow.apache.org/docs/python/extending.html#building-extensions-against-pypi-wheels. We had to make changes to prevent shared libraries from being duplicated in the wheels. It might be possible to change things to ship unversioned shared libraries instead, but it will require some surgery on the Cython modules to get them to link with the unversioned libs (because they look for the libraries with the ABI version when they link). Feel free to open a JIRA issue in Apache Arrow with a proposal -- I have definitely hit the limit for how much time I can personally spend on it

Reply:
Thanks @wesm

@timkpaine I am not particularly bothered by this, it is easy to fix and this file must be manually updated for new Arrow versions regardless.